freeipa-client/src/main/java/com/sequenceiq/freeipa/client/operation/SetWlCredentialOperation.java

```diff
@@ -38,8 +47,17 @@ public static SetWlCredentialOperation create(String user, String hashedPassword
             attributes.add("cdpHashedPassword=" + hashedPassword);
             attributes.add("cdpUnencryptedKrbPrincipalKey=" + unencryptedKrbPrincipalKey);
             attributes.add("krbPasswordExpiration=" + expiration);
+        }
+
+        // In order for the workload credentials update optimization to work correctly, we must update the title attribute
+        // whenever we set workload credentials in the IPA, whether or not a workload password is set. This is because the
+        // credentials version might have changed; for example, due to the addition or deletion of an SSH public key.
+        // However, when the optimization is disabled, we don't modify the title attribute at all.
+        title.ifPresent(titleAttr -> attributes.add("title=" + titleAttr));
```

Comment:
> why are we using the title? shouldn't we use a custom attribute? are we sure we won't need this field in freeipa in the future?

After some discussion with Kat, we believe that it's OK for us to take over an existing attribute, since we "own" the IPA/LDAP user record -- that is, we have full control over it, and it should not ever be modified by customers. This saves us the work of updating the schema, and of dealing with backward compatibility.

We aren't using 'title' for anything now, and it's hard to imagine that we would ever need to store an actual job title. The other advantage of 'title' is that it's in the default set of LDAP attributes searched by the IPA `user_find` command, which gives us the ability to search for a user by CRN easily.

There is some discussion related to this in this doc: https://docs.google.com/document/d/1jXhCruiM18cvfONoiQUrf37TW_pBhJpc-LweH_a4R1U/edit?usp=sharing

If you still have concerns, we should set up a meeting to discuss this with Kat and the CB and ID mgmt teams.